### PR TITLE
Added a boolean flag for disabling Entity onUpdate

### LINKED ISSUE #3694: [Feature request] Event for EntityTick
Currently, there is an event for EntityLivingUpdate. However, there is no similar event for all entities. The base entity class contains an onUpdate method, so why can't there be an event for onUpdate with any instance of Entity (not just EntityLivingBase)? Otherwise, the only way to, for example cancel a non-living entity update (like an arrow) is with ASM.

### PR DESCRIPTION
**Pull request by Da-Technomancer**
The flag is false by default, and if true the Entity will not update. It
is saved and loaded from NBT, and is called updateBlocked.
This would Close #3694 and acts in accordance to LexManos's suggestion in the discussion for that issue.

### CODE REVIEW

**Comment by williewillus** [comment]
put a comment so people at know what it does

**Comment by williewillus** [comment]
also isn't this going to fail for any entity that overrides onUpdate?

**Comment by LexManos** [comment]
Yes which is why the check should be in the place we have the event now, just less crappy event wise.

**Comment by Da-Technomancer** [comment]
I don't see what you mean by fail if it is overriden. Yes, overriding the method could stop the flag from having any effect, but overriding the method also can stop the event from having any effect in EntityLivingBase. Also, where should the comment go exactly? On the conditional in onUpdate or the declaration of updateBlocked?

**Comment by williewillus** [comment]
imo the flag checking should happen in world / whatever calls the Entity tick, so that it's always checked for every kind of entity, and you can't accidentally get rid of the check by overriding

**Comment by Da-Technomancer** [comment]
This commit adds a comment to the updateBlocked flag, and moves the updateBlocked check to the places that call onUpdate.

File: patches/minecraft/net/minecraft/world/World.java.patch [hunk 1]
```diff
@@ -493,7 +501,15 @@
          {
              p_72866_1_.field_70142_S = p_72866_1_.field_70165_t;
              p_72866_1_.field_70137_T = p_72866_1_.field_70163_u;
-@@ -1997,6 +2139,11 @@
+@@ -1816,6 +1959,7 @@
+                 }
+                 else
+                 {
++                	if(!p_72866_1_.updateBlocked)
```
Comment (mezz) [inline patches/minecraft/net/minecraft/world/World.java.patch]: you have a tab here

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by Da-Technomancer** [comment]
Tab replaced with 4 spaces.

**Review by mezz — APPROVED** [review]

**Comment by GotoLink** [comment]
I think this deserve some test case to show it doesn't break other behaviors of entities like loading/unloading, or collision.

**Comment by Da-Technomancer** [comment]
I've been using the test case. There don't seem to be any problems with loading/unloading or collision that I can find.

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 2]
```diff
@@ -9,7 +9,15 @@
  {
      private static final Logger field_184243_a = LogManager.getLogger();
      private static final List<ItemStack> field_190535_b = Collections.<ItemStack>emptyList();
-@@ -197,7 +197,7 @@
+@@ -175,6 +175,7 @@
+     private boolean field_184237_aG;
+     private double[] field_191505_aI;
+     private long field_191506_aJ;
++    public boolean updateBlocked;//Setting this to true will prevent the world from calling onUpdate for this entity.
```
Comment (HenryLoenwind) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch]: can you make this a javadoc comment? Then we'll see it on mouseover everwhere this field is used. Or would this make the patch too big?

Comment (mezz) [reply patches/minecraft/net/minecraft/entity/Entity.java.patch]: Not sure Henry. 
@LexManos what is the right way to document this new field, can it be a javadoc?

Comment (LexManos) [reply patches/minecraft/net/minecraft/entity/Entity.java.patch]: javadoc is fine,  especially if it's modder facing. Internal stuff with //FIXME, //TODO, or //REMOVE directed twards myself/cpw/fry are fine.

**Comment by Da-Technomancer** [comment]
Comment replaced with JavaDoc.

**Comment by mezz** [comment]
Looks good. Tested every vanilla entity and it works as intended.
There are some interesting side effect of blocking updates since everything is basically frozen in time, like zombies will not burn in daylight, and burning zombies will not take damage. This PR does what it's supposed to though.

**Pull request merged**